src/App.tsx

```diff
@@ -15,47 +16,135 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType !== SortType.NONE) {
+    visibleGoods.sort((firstGood, secondGood) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return firstGood.localeCompare(secondGood);
+        case SortType.LENGTH:
+          return firstGood.length - secondGood.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const goods = getReorderedGoods(goodsFromServer, { isReversed, sortType });
+
+  const handleSortChange = (newSortType: SortType) => {
+    setSortType(newSortType);
+  };
+
+  const handleReverseChange = () => {
+    setIsReversed(previous => !previous);
+  };
```

Comment:
```suggestion
  const handleSortChange = (newSortType: SortType) => {
    setSortType(newSortType);
  };

  const handleReverseChange = () => {
    setIsReversed(previous => !previous);
  };
```
why do you need extra function, you can use just hooks 
for example
onClick={() => setSortType(SortType.ALPHABET)}

Reply:
I use this function in 2 places, and it could potentialy have additional checks or other functionality. And my jsx looks cleaner